### PR TITLE
Splitting up global configs into pages

### LINKED ISSUE #3184: Split up the Global Configs page
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### What page(s) or areas on docs.getdbt.com are affected?

https://docs.getdbt.com/reference/global-configs

### What changes are you suggesting?

This page is very long and difficult to parse. Splitting it up into individual reference pages similar to the `List of Commands` pages. The `Global Configs` doc would become a parent category with the individual articles broken up below it. 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?
The `global-configs` reference guide had become a single page of increasing size. This breaks it down into individual pages which will make the content a little easier to find, especially when using the search functions.

Most of the features in the original one-shot get their own page but some gather multiple sections into a single page. Specifically:
`print-output`
`logs`
`parsing`

## Checklist
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [x] Add page to `website/sidebars.js`
- [x] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [x] Remove page from `website/sidebars.js`
- [x] Add an entry `website/static/_redirects`
- [x] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

**Review by jtcohen6 — COMMENTED** [review]
@matthewshaver Very supportive of splitting up this page into meaningful subsets, and I like the groupings you've chosen!

Meta-level comment: The sub-pages on "CLI flags," "env vars," and "yaml configs" should be presented differently. These aren't about **what** you can configure, but rather **how** you can define these configurations. IMO it makes sense to keep these together, possibly still as part of the top-level page.

**Comment by dbeatty10** [comment]
@matthewshaver Exciting that you are working on this!

The Netlify preview failed, so I could only look at the code and not the actual web pages.

There are a couple issues that would be awesome to address either in this PR or a follow-up:
- https://github.com/dbt-labs/docs.getdbt.com/issues/3264
- https://github.com/dbt-labs/docs.getdbt.com/issues/2894

**Comment by dbeatty10** [comment]
@matthewshaver how would you feel about converting this into a draft until it's ready for review? i.e., draft until it's able to build in Netlify, etc.

**Comment by matthewshaver** [comment]
@dbeatty10 Just wrapped up those pesky Netlify errors. It should pass on the next check and make the preview available!

File: website/docs/reference/commands/debug.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ sidebar_label: "debug"
 id: "debug"
 ---
 
-`dbt debug` is a utility function to test the database connection and show information for debugging purposes. Not to be confused with [debug-level logging](/reference/global-configs#debug-level-logging) via the `--debug` option which increases verbosity.
+`dbt debug` is a utility function to test the database connection and show information for debugging purposes. Not to be confused with [debug-level logging](/reference/about-global-configs#debug-level-logging) via the `--debug` option which increases verbosity.
```
Comment (matthewshaver) [inline website/docs/reference/commands/debug.md]:
```suggestion
`dbt debug` is a utility function to test the database connection and show information for debugging purposes. Not to be confused with [debug-level logging](/reference/global-configs/about-global-configs#debug-level-logging) via the `--debug` option which increases verbosity.
```

**Comment by mirnawong1** [comment]
approving this for Matt to merge, and he said he'll tackle Doug's feedback in separate pr

**Review by mirnawong1 — APPROVED** [review]
🚢 @matthewshaver

**Pull request merged**